### PR TITLE
Update to Celluloid 0.17

### PR DESCRIPTION
**Pull request by donaldpiret**
Not sure if this is 100% correct. Couldn't really figure out what the CellProxy was doing. Would appreciate if you could have a look.

### CODE REVIEW

**Comment by brandonhilkert** [comment]
Hey Donald! Thanks for the this. Give me a little bit to look over and understand what's going on with new Celluloid and I'll get back to you. Thanks again!

**Comment by eileencodes** [comment]
When you merge this, it needs to go to celluloid 0.17.2, there's apparently a memory leak in anything below 0.17.2 although the repo doesn't confirm this in the changelog. I read about it here https://github.com/mperham/sidekiq/issues/2582

**Comment by brandonhilkert** [comment]
I saw a similar Tweet from Mike yesterday. Thanks for the heads up!


------------------------------------
*Build a Ruby Gem is available!
<http://brandonhilkert.com/books/build-a-ruby-gem/?utm_source=gmail-sig&utm_medium=email&utm_campaign=gmail>*

http://brandonhilkert.com

File: sucker_punch.gemspec [hunk 1]
```diff
@@ -22,5 +22,5 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency "rake"
   gem.add_development_dependency "pry"
 
-  gem.add_dependency "celluloid", "0.16.0"
+  gem.add_dependency "celluloid", "~> 0.17.1", ">= 0.17.1.2"
```
Comment (brandonhilkert) [inline sucker_punch.gemspec]: Should we make this `"~> 0.17.2"`?

Comment (donaldpiret) [inline sucker_punch.gemspec]: Yup, thank you for handling that! Sorry I couldn't take care of it earlier. Ended up having a very busy last couple of days.

**Comment by brandonhilkert** [comment]
Release version `1.6.0` today as a result of your work. Thanks for the help here!

**Pull request merged**